lib/java-extras/src/main/java/org/triplea/java/exception/UnhandledSwitchCaseException.java

```diff
@@ -0,0 +1,24 @@
+package org.triplea.java.exception;
+
+/**
+ * Exception to throw when a default case block on a switch should never be executated.
+ *
+ * <p>EG: <code>
+ *   <pre>
+ *
+ *  switch(someValue) {
+ *    case ...:
+ *       break;
+ *    default:
+ *       throw new UnhandledSwitchCaseException(someValue);
+ *  }
+ *   </pre>
+ * </code>
+ */
+public class UnhandledSwitchCaseException extends RuntimeException {
```

Comment:
Didn't we use `AssertionError` for this in the past?

Reply:
Looks like. I don't think `AssertionError` quite fits the bill for that usage, having a more direct exception seems like an improvement - if anything just so we don't have to repeatedly write the boiler plate `"Unhandled case value: " + value`.

Reply:
IIRC the rationale behind it was that it essentially represented an impossible condition that if it would ever be reached would indicate an error in the jvm.
So throwing an error instead of an exception makes sense there.
However I think you could also argue that this kind of exception would indicate that the developer forgot to handle the case after altering the enum type, where a dedicated runtime exception with a good message would also make sense.
Depends on your standpoint though I think

Reply:
Well, both standpoints can be true : )
`AssertionError` -> true at the time of writing
`UnhandledSwitchCaseException` -> true after anyone modifies the enum